clang/lib/CConv/ConstraintVariables.cpp

```diff
@@ -957,6 +957,42 @@ bool PointerVariableConstraint::hasNtArr(EnvironmentMap &E)
   return false;
 }
 
+bool PointerVariableConstraint::
+    haveSameAssignment(Constraints &CS, ConstraintVariable *CV) {
+  bool Ret = false;
```

Comment:
Do we care about them having the same assignment, or do we just care if they are representing the same type? That is, it could be that one side is a `ConstAtom` and the other is a `VarAtom` but whose solution is to the same `ConstAtom`.

Reply:
Yes, the function actually checks if they are of the same type.

Reply:
So, maybe rename to `solutionEqualTo`? It's a subtle difference, but the name emphasizes that they don't have to have the same structure of `VarAtom`s --- as long as the two sides are equal, it doesn't matter how they started.